src/app/views/previous-routes/previous-routes.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit } from '@angular/core';
+import { Route } from 'src/app/models/route';
+import { DataService } from 'src/app/services/data.service';
+import { Vehicle } from 'src/app/models/vehicle';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { IconService } from 'src/app/services/icon.service';
+import { RouteService } from 'src/app/services/route.service';
+
+@Component({
+  selector: 'app-previous-routes',
+  templateUrl: './previous-routes.component.html',
+  styleUrls: ['./previous-routes.component.css']
+})
+export class PreviousRoutesComponent implements OnInit {
+
+  private modalRoute: Route;
+
+  constructor(
+    private dataService: DataService,
+    private iconService: IconService,
+    private routeService: RouteService,
+    private modalService: NgbModal) { }
+
+  open(content: any, route: Route): any {
+    this.modalRoute = route;
+
+    this.modalService.open(content, {ariaLabelledBy: 'modal-basic-title', centered: true});
+  }
+
+  ngOnInit(): void {
+  }
+
+  getRoutes(): Route[] {
+    return Object.values(this.dataService.getRoutes());
+  }
+
+  deleteRoute(route: Route): void {
+    this.dataService.deleteRoute(route);
+  }
+
+  validatePassengerInput(event: any, route: Route): void {
+    if (!event.target.validity.valid) {
```

Comment:
just a preference: I would put event.target into a variable at the top of the function